server/portal/apps/onboarding/steps/project_membership.py

```diff
@@ -61,12 +61,8 @@ def get_tracker(self):
     def is_project_member(self):
         username = self.user.username
         tas_client = self.get_tas_client()
-        projects = tas_client.projects_for_user(username)
-        return any(
-            [
-                project['id'] == self.settings['project_sql_id'] for project in projects
-            ]
-        )
+        project_users = tas_client.get_project_users(self.settings['project_sql_id'])
```

Comment:
Two questions

1. Do we know if `get_project_users` has pagination or any other upper limit in response items we might need to worry about?
2. I know this is a workaround for allocation-less projects. Do we know for sure that there isn't the case for a project-less allocation, or an allocation with a project of a different name?

Reply:
> Do we know if get_project_users has pagination or any other upper limit in response items we might need to worry about?

not aware but will check things out.  I'm guessing the answer is gonna be "no pagniation or limits in tas" as I haven't seen anything related to limits in pytas (https://bitbucket.org/taccaci/pytas/src/8055ede4d5cc4ed6a634fc3782ec8225e6784477/pytas/http.py#lines-346:352) .  but I'll test things out by using pytas for getting users on a large project like DS or something else like that.

Reply:
> I know this is a workaround for allocation-less projects. Do we know for sure that there isn't the case for a project-less allocation, or an allocation with a project of a different name?

i'm not following the question. gonna reach out on slack for clarifcation

Reply:
Question 1 
>> Do we know if get_project_users has pagination or any other upper limit in response items we might need to worry about?

> not aware but will check things out. I'm guessing the answer is gonna be "no pagniation or limits in tas" as I haven't seen anything related to limits in pytas (https://bitbucket.org/taccaci/pytas/src/8055ede4d5cc4ed6a634fc3782ec8225e6784477/pytas/http.py#lines-346:352) . but I'll test things out by using pytas for getting users on a large project like DS or something else like that.



 Using that sql id of 40095, I got got 2262 users (from  `get_project_users` ) for DesignSafe-DCV (https://tas.tacc.utexas.edu/TACCAdministration/Projects/EditProject.aspx?id=40095). That matches up with the number of users shown in TUP. 👍

Reply:
Discussed with Nathan, concerns resolved.